File: Cakezone/src/Models/Order.ts

export class Order{

    oId?:number;
    orderNumber:number;
    orderDate?:string;
    quantity:number;
    orderState?:string;
    totalPrice:number;
    product:any;
    customer:any;

    constructor(
        orderNumber:number,
        quantity:number,
        totalPrice:number,
        product:any,
        customer:any){

            this.orderNumber = orderNumber;
            this.quantity = quantity;
            this.totalPrice = totalPrice;
            this.product = product;
            this.customer = customer;
    }

}

File: cakezone-backend/src/main/java/com/lakshithafdo/cakezone_backend/Entity/Order.java
package com.lakshithafdo.cakezone_backend.Entity;


import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long oId;
    private long orderNumber;
    private String orderDate;
    private int quantity;
    private String orderState;
    private String totalPrice;

    @ManyToOne
    @JoinColumn (name = "fk_product")
    private Products product;


    @ManyToOne
    @JoinColumn (name = "fk_customer")
    private Customer customer;




}